### DISCUSSION TITLE
Global testcase run error

### DISCUSSION THREAD

**Discussion by hashmifatima** [NASA-LIS/LISF#877] [General]
Hi,

I am trying to run a testcase ~/LISF_557WW_release_7.4.2/lis/testcases/domains/gaussian_t126
but I am getting error. error log is attached here.
[lisconfig.docx](https://github.com/NASA-LIS/LISF/files/6966836/lisconfig.docx)

[lislog.docx](https://github.com/NASA-LIS/LISF/files/6966814/lislog.docx)
 
Kindly check. How to resolve this error.

Thanks,
Hashmi

**Comment by emkemp** [comment]
Hi @hashmifatima 

From your lis.config file, it appears you are using 4 processors.  Can you check the other three lislog files and see if any of them give more output, and preferably an error message?  Unfortunately there is a race condition with multi-processor runs, and one process may encounter the error and abort before the others do.

Also, assuming you ran this as a batch job with PBS, SLURM, or something similar:  Is there a standard out or standard error file from the batch job?  On our system, we sometimes see messages from a SLURM batch output file with additional information, like a stack trace.

**Reply by hashmifatima** [reply]
Hi @emkemp 

I am attaching all the log files here. I am running this as a batch job with PBS.

Most of the files I have found, but I am not able to find these two files:
Observation perturbation algorithm:       "none"
Observation perturbation frequency:       "6hr"
Observation attributes file:              ./LPRMobs_attribs.txt
Observation perturbation attributes file: ./LPRMobs_pertattribs.txt

GDAS data I have downloaded from here:  https://portal.nccs.nasa.gov/lisdata_pub/data/MET_FORCING/GDAS/200001/
 
Could you please suggest any other Global testcase which can be run by LIS 7.4.2? 
 
standard out and standard error file from the batch job:
[lis_mihir_log.docx](https://github.com/NASA-LIS/LISF/files/6973808/lis_mihir_log.docx)


[mylislog.docx](https://github.com/NASA-LIS/LISF/files/6973807/mylislog.docx)
[lislog0001.docx](https://github.com/NASA-LIS/LISF/files/6973766/lislog0001.docx)
[lislog0002.docx](https://github.com/NASA-LIS/LISF/files/6973767/lislog0002.docx)
[lislog0003.docx](https://github.com/NASA-LIS/LISF/files/6973768/lislog0003.docx)
[lislog0000.docx](https://github.com/NASA-LIS/LISF/files/6973769/lislog0000.docx)

**Comment by emkemp** [comment]
Hi @hashmifatima 

The key is in mylislog, which has a stacktrace and the error message "SIGSEGV, segmentation fault occurred".  Segmentation faults can be triggered by insufficient stack memory.

If you are using bash, make sure to set 'ulimit -s unlimited' in your batch job before invoking LIS.  (I don't know the equivalent for tcsh.)

**Reply by hashmifatima** [reply]
Hi @emkemp 

1. I am attaching here my LIS job script.  I already had 'ulimit -s unlimited' in my lis job scr
[lis_mihirjob.docx](https://github.com/NASA-LIS/LISF/files/6975737/lis_mihirjob.docx)
ipt.

2. Where I will find these files 'LPRMobs_attribs.txt' and 'LPRMobs_pertattribs.txt'?

Regards,
Hashmi

**Reply by dmocko** [reply]
Hello,

It's been a while since I used PBS, but I noticed that the number of CPUs in your PBS settings don't match the number of CPUs you requested in your aprun.

#PBS -l select=4:ncpus=36

aprun -n 4 -N 1 -d 1 ./LIS -f lis.config >mylis.log 2>&1

You could try to make them consistent to see if the error repeats.

Also, we generally recommend that non-557WW team members use the public releases.  The most recent public release is 7.3.1:
    https://github.com/NASA-LIS/LISF/releases/tag/v7.3.1-public

About your second question - we don't distribute many attribute files.  You must create them yourself using the LIS Users' Guide as a reference.

David

**Reply by emkemp** [reply]
Hmm, I'm looking at instructions for aprun on another site (https://bluewaters.ncsa.illinois.edu/using-aprun), and it looks like the settings used by @hashmifatima are correct.

-n: Number of processing elements PEs required for the application.  Set to 4, which matches the product of the numbers of processors in the x and y directions set in lis.config.
-N: Number of PEs to place per node.  So only 1 processor is used on each node.
-d: Number of CPU cores required for each PE and its threads.  Again, only 1 per PE.

I'm assuming ncpus=36 means "pick nodes with 36 physical cores each", not that all 36 cores are to be used.

I wonder if the OMP_STACKSIZE=2G is artificially lowering the stack size.  @hashmifatima, can you try changing this to 4G and see if the simulation goes further?

Another comment:  Use of the 557ww releases is allowed for UM partners, though they need to get the special tarfiles we put together if they want to use JULES.

**Comment by emkemp** [comment]
Hi @hashmifatima 

We asked around and found another LIS user who has assimilated AMSRE-LPRM soil moisture data, and who kindly shared his attribute files.  Note that the settings in these files were customized for the NCA-LDAS domain, described by Kumar et al., J Hydrometeor, 2018, https://doi.org/10.1175/JHM-D-17-0125.1
 
[AMSRE_attribs.txt](https://github.com/NASA-LIS/LISF/files/6977653/AMSRE_attribs.txt)
[AMSRE_pertattribs.txt](https://github.com/NASA-LIS/LISF/files/6977654/AMSRE_pertattribs.txt)

You likely will need to tune these settings to your domain, but hopefully they will be a useful first guess.

**Comment by hashmifatima** [comment]
Hi @emkemp  

I have been struggling with the global testcase "gaussian_t126", but it was giving errors.
So I thought to try with Global use case with Jules.5.0 which you have provided during LIS tutorial. So I copied ldt.config and lis.config and the data files. 

LDT run was successful. I was able to create LIS input data file "lis_input.global.jules50.nc". But LIS run is giving errors with AGRMET forcing data. I am attaching  lis.config file. I have copied lis.config from "lis.config.from.jerry".
[mylis.log](https://github.com/NASA-LIS/LISF/files/7039072/mylis.log)
[lis.config.fromhashmi.docx](https://github.com/NASA-LIS/LISF/files/7039097/lis.config.fromhashmi.docx)
[lismihir1D.docx](https://github.com/NASA-LIS/LISF/files/7039105/lismihir1D.docx)
[logs.tar.gz](https://github.com/NASA-LIS/LISF/files/7039106/logs.tar.gz)

forcing directory is same as given with Global_use_case tar.
Kindly check lis.config and error logs.

~
Hashmi

**Reply by emkemp** [reply]
This is being handled via e-mail, and cc:'ed to other Met Office partners.
